submissions/BlueLamantine/task-js-opp/app.js

```diff
@@ -0,0 +1,100 @@
+const WIDTH = 505;
+const BLOCK_HEIGHT = 77;
+const STEP = 90;
+const MAX_SPEED = 400;
+const MIN_SPEED = 100;
+const PLAYER_CONF = {
+    startPosX : 200, 
+    startPosY : 400,
+};
+
+const ENEMY_CONF = {
+    quantity : 3,
+    startPosX : -100,
+    startPosY : [1,2,3].map(rowNumber => rowNumber * BLOCK_HEIGHT),
+};
+
+let Enemy = function(x,y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/cat.gif';
+};
+
+let allEnemies =  Array.from(Array(ENEMY_CONF.quantity),
+(val, index) => new Enemy(ENEMY_CONF.startPosX,ENEMY_CONF.startPosY[index],getRandomNumber(MIN_SPEED,MAX_SPEED)));
+
+Enemy.prototype.update = function(dt) {
+    this.x > WIDTH ? this.x = ENEMY_CONF.startPosX : this.x = this.x + this.speed * dt;
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+let Player = function() {
+    this.x = PLAYER_CONF.startPosX;
+    this.y = PLAYER_CONF.startPosY;
+    this.sprite = 'images/fish.gif';
+};
+Player.prototype.setDefault = function () {
+    this.x = PLAYER_CONF.startPosX;
+    this.y = PLAYER_CONF.startPosY;
+};
+
+Player.prototype.update = function() {
+    if (this.y < 0) {
+        alert('WINNER, you saved the fish!');
+        player.setDefault();
+    }
+   // handles collision with the Player
+   allEnemies.forEach(function(enemy) {
+    if ( Math.round(enemy.x / STEP) === Math.round(player.x / STEP) && 
+         Math.round(enemy.y / BLOCK_HEIGHT) === Math.round(player.y / BLOCK_HEIGHT) ) {
+            alert('you lose! =(');
+            player.setDefault();
+    }
+   });
```

Comment:
If you had to add a comment this is a clear sign that this part is worth being isolated into a function/method.
Besides `update` does two very diifferent things : check location and check collisions. This is violation of the SIngle Responsibility Principle.

Solution:
Define a dedicated function to check against collisions.
Call it from `update`

Reply:
Thanks for comment, my bad. I fixed it according to your proposed solution!